consensus/consensus-types/src/vote_data.rs

```diff
@@ -40,6 +40,12 @@ impl VoteData {
         &self.proposed
     }
 
+    /// Returns true if the vote is on a NIL block
+    pub fn is_nil(&self) -> bool {
+        self.proposed.timestamp_usecs() == self.parent.timestamp_usecs()
```

Comment:
so this is wrong, another way I can check this is `proposed.version = parent.version + 1`. That should work all the time IIUC (cc @zekun000)

Reply:
no, a nil block can be reconfiguration suffix too which have the same version

Reply:
so I would have to filter any block that has more than one version added to it for this PR to work... I feel like I'm missing something about block metadata though, I'll dig a bit more.